platform/android/os_android.cpp

```diff
@@ -538,6 +538,20 @@ String OS_Android::get_system_dir(SystemDir p_dir, bool p_shared_storage) const
 	return godot_io_java->get_system_dir(p_dir, p_shared_storage);
 }
 
+Error OS_Android::move_to_trash(const String &p_path) {
+	Error err = FAILED;
+	DirAccessRef da_ref = DirAccess::open(p_path, &err);
+	if (!da_ref) {
+		return err;
+	}
+
+	err = da_ref->erase_contents_recursive();
+	if (err) {
+		return err;
+	}
+	return da_ref->remove(p_path);
+}
```

Comment:
The logic on the java side checks whether it's a file or a directory and apply the proper deletion mechanism accordingly.
See https://github.com/godotengine/godot/blob/3.x/platform/android/java/lib/src/org/godotengine/godot/io/directory/FilesystemDirectoryAccess.kt#L216-L222

Reply:
That's for the call to `remove()`, but the `DirAccess::open()` and `erase_contents_recursive()` would fail I think?

And if `open()` actually succeeds, or falls back to `res://`, it will attempt deleting contents that the user didn't request.

Reply:
> That's for the call to `remove()`, but the `DirAccess::open()` and `erase_contents_recursive()` would fail I think?
> 
> And if `open()` actually succeeds, or falls back to `res://`, it will attempt deleting all folder contents that the user didn't request.

Yes, that's right. I've updated the logic and provided separate branches for directories and files handling.